ui/pages/permissions-connect/snaps/snaps-connect/snaps-connect.js

```diff
@@ -57,7 +58,12 @@ export default function SnapsConnect({
   const snaps = getDedupedSnaps(request, currentPermissions);
 
   const SnapsConnectContent = () => {
-    const { hostname: trimmedOrigin } = useOriginMetadata(origin) || {};
+    let trimmedOrigin = (useOriginMetadata(origin) || {})?.hostname;
+    const { name } = useSelector((state) => getSnapMetadata(state, origin));
```

Comment:
Hmm, this hook is a bit problematic since it'll crash when `origin` is not a Snap.

Reply:
[736aebd](https://github.com/MetaMask/metamask-extension/pull/26715/commits/736aebd8d8930f97f908ffcd0c51c44e35bd2852)